### PR TITLE
fix(AutoLayout): collapsed child would still contribute to item spacing

### LINKED ISSUE #1026: When using Spacing + children Visible = Collapse, autolayout takes wrong size
## Current behavior

When we add many children to an AutoLayout with spacing > 0 and many of those children have visibility set to collapsed, the total size of the AutoLayout counts the sum of the spacings even between invisible children. 

## Expected behavior

IMHO, it should ignore both: collapsed children AND its spacing. 

## How to reproduce it (as minimally and precisely as possible)

https://playground.platform.uno/#46d88139

Nuget Package:

Package Version(s):

Affected platform(s):

- [ ] WebAssembly
- [ ] Android
- [ ] iOS
- [ ] macOS (AppKit)
- [ ] Mac Catalyst
- [ ] Skia
  - [ ] WPF
  - [ ] GTK (Linux)
  - [ ] Linux Framebuffer
  - [ ] Tizen
- [ ] Windows

IDE:

- [ ] Visual Studio 2022
- [ ] Visual Studio 2019
- [ ] Visual Studio Code
- [ ] Visual Studio for Mac
- [ ] Rider Windows
- [ ] Rider macOS

Relevant plugins:

### PR DESCRIPTION
**Pull request by Xiaoy312**
GitHub Issue (If applicable): closes #1026, closes unoplatform/dispatchscience-private#30

## PR Type
What kind of change does this PR introduce?
- Bugfix

## What is the new behavior?
Collapsed child views under AutoLayout no longer contribute to the total item spacing.

## PR Checklist
Please check if your PR fulfills the following requirements:
- [x] Tested code with current [supported SDKs](../README.md#supported)
- [x] Tested the changes where applicable:
	- [x] Desktop
	- [ ] WinUI
	- [ ] iOS
	- [ ] Android
	- [ ] WASM
	- [ ] MacOS
- [ ] Updated the documentation as needed:
	- [ ] [General Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc)
	- [ ] [Controls Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/controls)
	- [ ] [Extensions Doc Update](https://github.com/unoplatform/uno.toolkit.ui/tree/main/doc/helpers)
	- [ ] [controls-styles.md](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/controls-styles.md)
	- [ ] [lightweight-styling.md (LightWeight Styling Resource Keys)](https://github.com/unoplatform/uno.toolkit.ui/blob/main/doc/lightweight-styling.md)
- [x] [Runtime Tests and/or UI Tests](https://platform.uno/docs/articles/contributing/guidelines/creating-tests.html) for the changes have been added (for bug fixes / features) (if applicable)
- [x] Contains **NO** breaking changes
- [x] Associated with an issue (GitHub or internal)
- [x] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

### CODE REVIEW

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs [hunk 1]
```diff
@@ -168,6 +168,11 @@ private void ApplyMinMaxValues(ref Size desiredSize)
 				role = AutoLayoutRole.Independent;
 				numberOfStackedChildren--;
 			}
+			else if (child is FrameworkElement { Visibility: Visibility.Collapsed })
+			{
+				role = AutoLayoutRole.Collapsed;
+				numberOfStackedChildren--;
```
Comment (agneszitte) [inline src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs:174]:
```suggestion
				// Collapsed children do not participate in layout or spacing.
				role = AutoLayoutRole.Collapsed;
				numberOfStackedChildren--;
```

Comment (agneszitte) [reply src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs:174]: Also maybe `if (numberOfStackedChildren > 0) numberOfStackedChildren--;` if there’s any chance this could underflow in edge cases

File: src/Uno.Toolkit.RuntimeTests/Tests/AutoLayoutTest.cs [hunk 2]
```diff
@@ -62,6 +62,66 @@ public async Task When_Collapsed()
 		layoutRect1Actual.Y.Should().Be(20);
 	}
```
Comment (agneszitte) [inline src/Uno.Toolkit.RuntimeTests/Tests/AutoLayoutTest.cs:64]: The new tests look good 👍, they already cover the vertical / middle child collapsed cases (both initially and late).  

A few more additions worth considering to strengthen coverage if not already present, I think:

- **Horizontal orientation**: Add mirrored tests for `Orientation.Horizontal` (initially and late collapse).  
- **First/last collapsed**: Verify no spacing is left when the first or last child is collapsed.  
- **All collapsed**: Container size should resolve to zero (or minimum) with no spacing.  
- **Toggle visibility**: Collapsing then restoring to `Visible` should restore layout/spacing properly.  
- **Margins**: Ensure margins of a collapsed child do not contribute to spacing.  
- **Nested AutoLayout**: A collapsed child that itself is an `AutoLayout` should behave consistently.

File: src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs (hunk 1 above)
Comment (Xiaoy312) [reply src/Uno.Toolkit.UI/Controls/AutoLayout/AutoLayout.Layouting.Measure.cs:174]: no chance of underflowing
1. we are for-looping on the same count value:
```
		var numberOfStackedChildren = children.Count;
		for (var i = 0; i < children.Count; i++)
```
2. we are only decrementing ever once per iteration

even if we ever are, then there is clearly a logic error somewhere, and we shouldn't hiding that with a dont-decrement-past-0 approach. better to let that blow up big

File: src/Uno.Toolkit.RuntimeTests/Tests/AutoLayoutTest.cs (hunk 2 above)
Comment (carldebilly) [reply src/Uno.Toolkit.RuntimeTests/Tests/AutoLayoutTest.cs:64]: Copilot can help a lot here to create those tests :-)
